azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -322,82 +369,78 @@ private WebAppConfiguration getRuntimeConfiguration(WebAppConfiguration configur
+                Log.warn(String.format("Invalid webContainer '%s' for java version: %s",
+                        this.getRuntime().getWebContainerRaw(), javaVersion));
+            } else {
+                defaultLinuxRuntimeStack = validRuntStackFromExistingConfiguration;
+            }
+        } else {
+            if (Objects.isNull(findStringInCollectionIgnoreCase(validRuntimeStacks, defaultLinuxRuntimeStack))) {
+                Log.warn(String.format("'%s' is not supported in java version: %s", defaultLinuxRuntimeStack, javaVersion));
+                defaultLinuxRuntimeStack = WebAppConfiguration.DEFAULT_LINUX_WEB_CONTAINER;
+            }
         }
-        final String defaultLinuxRuntimeStack = configuration.getLinuxRuntimeStackOrDefault();
-        final String runtimeStack = queryer.assureInputFromUser("runtimeStack",
-            defaultLinuxRuntimeStack, RuntimeStackUtils.getValidWebContainer(javaVersion), null);
+        final String runtimeStack = queryer.assureInputFromUser("runtimeStack", defaultLinuxRuntimeStack,
+                validRuntimeStacks, null);
         return builder.runtimeStack(RuntimeStackUtils.getRuntimeStack(javaVersion, runtimeStack));
     }
 
     private WebAppConfiguration.Builder getRuntimeConfigurationOfWindows(WebAppConfiguration.Builder builder,
-                                                                         WebAppConfiguration configuration)
-        throws MojoFailureException {
-        final String defaultJavaVersion = configuration.getJavaVersionOrDefault();
-        final String javaVersionInput = queryer.assureInputFromUser("javaVersion", defaultJavaVersion, getAvailableJavaVersion(), null);
-        final JavaVersion javaVersion = JavaVersion.fromString(javaVersionInput);
+            WebAppConfiguration configuration) throws MojoFailureException {
+        String defaultJavaVersion = JavaVersionUtils.formatJavaVersion(configuration.getJavaVersionOrDefault());
+        final List<String> validJavaVersions = JavaVersionUtils.getValidJavaVersions();
+        if (!validJavaVersions.contains(defaultJavaVersion)) {
+            Log.warn(String.format("'%s' is not supported.", defaultJavaVersion));
+            defaultJavaVersion = WebAppConfiguration.DEFAULT_LINUX_JAVA_VERSION;
+        }
+
+        final String javaVersionInput = queryer.assureInputFromUser(JAVA_VERSION, defaultJavaVersion,
+                validJavaVersions, String.format(COMMON_PROMPT, JAVA_VERSION, defaultJavaVersion));
+        final JavaVersion javaVersion = JavaVersionUtils.toAzureSdkJavaVersion(javaVersionInput);
         // For project which package is jar, use java se runtime
         if (isJarProject()) {
-            return builder.javaVersion(javaVersion).webContainer(getJavaSEWebContainer(javaVersion));
+            final WebContainer webContainer = WebContainerUtils.getJavaSEWebContainer();
+            if (Objects.isNull(webContainer)) {
```

Comment:
`WebContainerUtils.getJavaSEWebContainer()` will return `WebContainer.fromString(JAVA_SE)` directly, will it be null sometime?

Reply:
this code is copied from:

 private static WebContainer getJavaSEWebContainer(JavaVersion javaVersion) throws MojoFailureException {	
        final String version = javaVersion.toString();	
        if (StringUtils.startsWith(version, JAVA_1_8)) {	
            return WebContainer.JAVA_8;	
        } else if (StringUtils.startsWith(version, JAVA_11)) {	
            return WebContainer.fromString(JAVA_11_STRING);	
        } else {	
            throw new MojoFailureException(String.format("Java SE environment is not supported in %s", javaVersion.toString()));	
        }	
    }	    }


since the java version is removed from webContainer , it is OK to remove this error.